tests/static_multiset/large_input_test.cu

```diff
@@ -46,7 +46,7 @@ void test_unique_sequence(Set& set, typename Set::value_type* res_begin, std::si
       set.retrieve(keys_begin, keys_end, thrust::make_discard_iterator(), res_begin);
     REQUIRE(static_cast<std::size_t>(std::distance(res_begin, res_end)) == num_keys);
 
-    thrust::sort(res_begin, res_end);
+    thrust::sort(thrust::device, res_begin, res_end);
```

Comment:
without this change, `thrust::sort` will fail on my local RTX8000.
```bash
yunsongw@0c23fdd-lcedt:~/Work/cuCollections/build/tests$ ./STATIC_MULTISET_TEST 
Randomness seeded to: 1468028837

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
STATIC_MULTISET_TEST is a Catch2 v3.3.0 host application.
Run with -? for options

-------------------------------------------------------------------------------
cuco::static_multiset large input test - int64_t, cuco::test::probe_sequence::
double_hashing, 1
  All inserted keys can be retrieved.
-------------------------------------------------------------------------------
/home/yunsongw/Work/cuCollections/tests/static_multiset/large_input_test.cu:43
...............................................................................

/home/yunsongw/Work/cuCollections/tests/static_multiset/large_input_test.cu:43: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGSEGV - Segmentation violation signal

===============================================================================
test cases:  57 |  56 passed | 1 failed
assertions: 487 | 486 passed | 1 failed

Segmentation fault (core dumped)
```

Will open an CCCL issue later.